### PR TITLE
Ingress Gateways integration with Agentless and Consul Dataplane

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Register Ingress Gateways using the endpoints controller.
- Use Consul Dataplane instead of Envoy

How I've tested this PR:
- Acceptance tests
- Unit and BATS tests.

How I expect reviewers to test this PR:
- Code Review

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/ingress-gateways-deployment.yaml [hunk 1]
```diff
@@ -107,374 +146,165 @@ spec:
...
+          requests:
+            memory: "50Mi"
+            cpu: "50m"
+          limits:
+            memory: "50Mi"
+            cpu: "50m"
+      containers:
+      - name: ingress-gateway
+        image: {{ $root.Values.global.imageConsulDataplane | quote }}
+        {{- if (default $defaults.resources .resources) }}
```
Comment (t-eckert) [inline charts/consul/templates/ingress-gateways-deployment.yaml:217]:
```suggestion
        {{- if (default $defaults.resources $root.resources) }}
```
Does the non-default value need to be pinned to `$root` here and below?

**Review by t-eckert — APPROVED** [review]
Looks great!

File: charts/consul/templates/ingress-gateways-deployment.yaml (hunk 1 above)
Comment (thisisnotashwin) [reply charts/consul/templates/ingress-gateways-deployment.yaml:217]: Its only those that start with a `.`

**Comment by thisisnotashwin** [comment]
Closing in favor of https://github.com/hashicorp/consul-k8s/pull/1544 to avoid rebase hell.

**Pull request closed without merging**